### DISCUSSION TITLE
Can someone explain how static site can call mongo at runtime ?

### DISCUSSION THREAD

**Discussion by RayLuxembourg** [vercel/next.js#20416] [Help]
I saw a video of someone creating [Incremental Static Generation and Regeneration](https://www.youtube.com/watch?v=IJkTpR7sSwI)
I cannot help to think, statically generate nextjs app which hosten on a S3, how is it calling a MongoDB O_O?

**Comment by Tomburgs** [comment]
Not entirely sure what's confusing you.
The database fetch calls are executed server-side, and every time it does the static content gets updated.

**Reply by RayLuxembourg** [reply]
@Tomburgs what server-side? the application is exported and hosted on s3 there is no server O_O

**Reply by Tomburgs** [reply]
You obviously cannot use s3 buckets, as you’d have to host the node process.

Something like ec2 should be used instead.

**Reply by RayLuxembourg** [reply]
I have been using next in production for the last 2 years. I was just wondering what's the deal with all the static stuff.
